src/node/txreconciliation.h

```diff
@@ -76,6 +76,36 @@ class TxReconciliationTracker
     ReconciliationRegisterResult RegisterPeer(NodeId peer_id, bool is_peer_inbound,
                                               uint32_t peer_recon_version, uint64_t remote_salt);
 
+    /**
+     * Step 1. Add new transactions we want to announce to the peer to the local reconciliation set
+     * of the peer, so that those transactions will be reconciled later.
+     * Returns the number of added transactions (already present transactions won't be added).
+     *
+     * The caller *must* check that the peer is registered for reconciliations.
+     */
+    size_t AddToSet(NodeId peer_id, const std::vector<uint256>& txs_to_reconcile);
+
+    /**
+     * Before Step 2, we might want to remove a wtxid from the reconciliation set, for example if
+     * the peer just announced the transaction to us.
+     * Returns whether the wtxid was removed.
+     *
+     * The caller *must* check that the peer is registered for reconciliations.
+     */
+    bool TryRemovingFromSet(NodeId peer_id, const uint256& wtxid_to_remove);
+
+    bool IsPeerNextToReconcileWith(NodeId peer_id, std::chrono::microseconds now);
```

Comment:
Can add a comment about what `TxReconciliationTracker::IsPeerNextToReconcileWith()` does, like assumptions on peers queue and phase we're in.